platform-common/src/main/java/com/softicar/platform/common/io/serialization/json/JsonValueReader.java

```diff
@@ -0,0 +1,88 @@
+package com.softicar.platform.common.io.serialization.json;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import com.softicar.platform.common.core.utils.DevNull;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Facilitates reading values from a JSON {@link String}.
+ * <p>
+ * Based upon "JsonPath" and its syntax.
+ *
+ * @author Alexander Schmidt
+ */
+public class JsonValueReader {
+
+	private final DocumentContext context;
+
+	/**
+	 * Constructs a new {@link JsonValueReader}.
+	 *
+	 * @param json
+	 *            the JSON {@link String} to process (never <i>null</i>)
+	 */
+	public JsonValueReader(String json) {
+
+		Objects.requireNonNull(json);
+		var configuration = Configuration.defaultConfiguration().setOptions(Option.SUPPRESS_EXCEPTIONS);
+		this.context = JsonPath.using(configuration).parse(json);
+	}
+
+	/**
+	 * Reads a {@link String} value from the given path inside the JSON
+	 * {@link String}.
+	 * <p>
+	 * Returns an empty {@link Optional} if no such path exists inside the JSON
+	 * {@link String}, or if the path points to an array of values.
+	 *
+	 * @param path
+	 *            the path, in "JsonPath" syntax (never <i>null</i>)
+	 * @return the {@link String} value to which the given path points
+	 */
+	public Optional<String> read(String path) {
```

Comment:
maybe `readValue ` would be a better name to constrast it to the `readList` method?

Reply:
ok